pkgs/by-name/gq/gqlgen/package.nix

```
@@ -0,0 +1,39 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildGoModule,
+}:
+
+let
+  version = "0.17.63";
+in
+buildGoModule {
+  pname = "gqlgen";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "99designs";
+    repo = "gqlgen";
+    tag = "v${version}";
+    hash = "sha256-J9+pleHdbQMHP/Aq9Pl6ise6PDvRqxQ72Iq7SNxgMws=";
+  };
+
+  vendorHash = "sha256-hPUWYOfCx+kW2dJsjkCE/7bwofnGdQbDTvfZ877/pCk=";
+
+  subPackages = [ "." ];
+
+  env.CGO_ENABLED = 0;
+
+  checkFlags = [
+    "-skip=^TestGenerate$" # skip tests that want to run `go mod tidy`
+  ];
+
```

[versionCheckHook](https://nixos.org/manual/nixpkgs/stable/#versioncheckhook) would be a nice addition here as a quick way to make sure the binary is working

```suggestion

  nativeInstallCheckInputs = [
    versionCheckHook
  ];
  doInstallCheck = true;

  versionCheckProgramArg = "version";

```

---

Thank you! Done.